### ISSUE TITLE
How to use inside of a Lambda?

### ISSUE THREAD

**Issue by kpeters-cbsi** [aspecto-io/sns-sqs-big-payload#12]
I'd like to use this inside of a Lambda that's triggered from an SQS queue. How would I do that? I don't think polling is an appropriate solution here as AWS has already provided me the SQS message.

**Comment by kpeters-cbsi** [comment]
I tried the following to no avail.
```
const sqsConsumer = SqsConsumer.create({
  queueUrl,
  region,
  getPayloadFromS3: true,
  transformMessageBody: (body) => JSON.parse(body),
  handleMessage: async ({ payload }) => {
    console.debug({ payload })
  }
}
sqsConsumer.start()
sqsConsumer.on(SqsConsumerEvents.batchProcessed, (...args) => {
  console.debug({ args })
  sqsConsumer.stop()
}
```

**Comment by jackvhall** [comment]
I have the same use case.

**Comment by mzahor** [comment]
Hi @jackvhall, @kpeters-cbsi. It's an interesting use case. We may implement it at some point, but currently we don't have it in our environment, so it's not a priority. However you can submit a PR. I think it may require some restructuring of the consumer's code, so let's discuss the design before actually implementing it.

**Comment by mzahor** [comment]
@jackvhall @kpeters-cbsi I just took a quick look at the code, seems like it's quite easy to achieve w/o any major changes. I've submitted a PR, but I didn't have chance to test it yet. Can u please check if it works for you?

**Comment by jackvhall** [comment]
@mzahor Thanks so much. I will take a look at the PR at some point tomorrow. At a glance, I think the idea and api change is right. We simply need to be able to pass the `event.Record` item directly to `sqsConsumer.processMessage` and have it handled normally beyond that point.

I hope that I can test it out tomorrow afternoon. Thanks so much for your work on this!

**Comment by xspecm** [comment]
@jackvhall 

Hi, can you tell if you have tested the new functionality without any major problems?

**Comment by jackvhall** [comment]
@xspecm @mzahor 

My team decided that the particular problem we were trying to solve was not worth pursuing in the existing application. We've chosen to leverage another mechanism more appropriate to our use case. However, I did manage to test out the functionality using the proposed changes in the PR and it does in fact work (provided the access modifier on `sqsConsumer.processMsg` is public).

Essentially, the ability to publicly call `sqsConsumer.processMsg` and bypass the deleting of the message from the queue (which only applies in a polling scenario) is all that is required to make this work for people accepting queue messages via trigger. I really appreciate the great work you guys have done on this package. Hopefully you will integrate this into the package as I am sure there are many people trying to solve this issue.

Thanks again.

**Comment by mzahor** [comment]
@jackvhall Thanks for checking it. I've released a new version (0.0.11) with the changes and the usage guide: https://github.com/aspecto-io/sns-sqs-big-payload/blob/master/docs/usage-in-lambda.md
